code/game/mob/living/carbon/human/human_defense.dm

```diff
@@ -628,15 +632,27 @@ bullet_act
 	else if (hit_zone == "l_leg" || hit_zone == "r_leg" || hit_zone == "r_arm" || hit_zone == "l_arm")
 		if (prob(25))
 			if (prob(60))
-				visible_message("<span class='notice'>[user] has tried to strike [src]'s [affecting.name] with [I.name] but missed!</span>")
-				return
+				if (user != src)
```

Comment:
Same here, no need to make it more complex

Reply:
No, this checks if the person is attacking themselves instead of [src], it's meant to be from different views of the clients to make it more immersive.